src/aics-image-viewer/components/ViewerStateProvider/index.tsx

```diff
@@ -76,52 +77,61 @@ const viewerSettingsReducer = <K extends keyof ViewerState>(
   }
 };
 
-/** Utility type to explicitly assert that one or more properties will *not* be defined on an object */
-type WithExplicitlyUndefined<K extends keyof any, T> = T & { [key in K]?: never };
+enum ChannelSettingActionType {
+  UniformUpdate = "UniformUpdate",
+  ArrayUpdate = "ArrayUpdate",
+  Init = "Init",
+}
```

Comment:
this file has the important change.  The actions now each have an action type, and they take `Record<K, ChannelState[K]>` which looks like `{key: value, key2:value2,...}` whereas they used to take only a single  `key:K, value:ChannelState[K]`